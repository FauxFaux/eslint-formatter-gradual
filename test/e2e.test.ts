import { spawnSync } from 'child_process';
import escapeStringRegexp from 'escape-string-regexp';

test('e2e (bin)', async () => {
  const out = spawnSync(process.argv0, ['../../dist/eslint-gradual', '*.mjs'], {
    cwd: 'test/e2e',
    stdio: 'pipe',
  });
  if (out.error) {
    throw out.error;
  }
  expect(cleanupStdout(out)).toMatchSnapshot();
  expect(out.status).toBe(1);
});

test('e2e (formatter)', async () => {
  const out = spawnSync(
    process.argv0,
    ['../../node_modules/.bin/eslint', '-f', '../..', '*.mjs'],
    {
      cwd: 'test/e2e',
      stdio: 'pipe',
    },
  );
  if (out.error) {
    throw out.error;
  }
  expect(cleanupStdout(out)).toMatchSnapshot();
  expect(out.status).toBe(1);
});

function cleanupStdout(out: { stdout?: Buffer }) {
  return out.stdout
    ?.toString('utf-8')
    .replace(new RegExp(escapeStringRegexp(process.cwd()), 'g'), '<rootDir>');
}
