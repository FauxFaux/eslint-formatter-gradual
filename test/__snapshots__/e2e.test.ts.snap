// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e (bin) 1`] = `
"
<rootDir>/test/e2e/one.mjs
     1:10   warning (ignored until touched)  'foo' is defined but never used  no-unused-vars
     4:5    warning (ignored for file)       Duplicate key 'a'                no-dupe-keys

<rootDir>/test/e2e/three.mjs
     2:10   error    'three' is defined but never used  no-unused-vars
     5:5    error    Duplicate key 'c'                  no-dupe-keys

<rootDir>/test/e2e/two.mjs
     1:10   error   (was ignored until touched)  'two' is defined but never used  no-unused-vars
     4:5    warning (ignored for file)           Duplicate key 'b'                no-dupe-keys

✖ 6 problems (3 errors, 3 warnings)

"
`;

exports[`e2e (formatter) 1`] = `
"
<rootDir>/test/e2e/one.mjs
     1:10   warning (ignored until touched)  'foo' is defined but never used  no-unused-vars
     4:5    warning (ignored for file)       Duplicate key 'a'                no-dupe-keys

<rootDir>/test/e2e/three.mjs
     2:10   error    'three' is defined but never used  no-unused-vars
     5:5    error    Duplicate key 'c'                  no-dupe-keys

<rootDir>/test/e2e/two.mjs
     1:10   error   (was ignored until touched)  'two' is defined but never used  no-unused-vars
     4:5    warning (ignored for file)           Duplicate key 'b'                no-dupe-keys

✖ 6 problems (3 errors, 3 warnings)

"
`;
